### PR TITLE
receiver/hostmetrics/disk: add serial number to disk metrics

### PR DESCRIPTION
**Pull request by gdvalle**
**Description:** <Describe what has changed.>

This adds a `serial_number` attribute to the hostmetricsreceiver's disk metrics. This data is already fetched as part of the [IOCountersStat](https://pkg.go.dev/github.com/shirou/gopsutil/disk#IOCountersStat) struct we use for other disk metrics, so nothing new is introduced.

Because gopsutil's serial number enumeration was broken when called from `IOCounters()`, this relies on [a patch in upstream](https://github.com/shirou/gopsutil/pull/1508) that's not yet in a release, so this also bumps gopsutil to a commit with that included.

The use case is associating disk metrics with cloud provider disks. For example, in Google Cloud, a serial number can look like `0Google_PersistentDisk_pvc-xxx`, where `pvc-xxx` is the GCP disk resource name. It's also the "short" serial number, which might be ideal, but gathering that would be larger patch for gopsutil, so I think we can settle for this.

**Link to tracking Issue:** <Issue number if applicable>
N/A

**Testing:** <Describe what testing was performed and which tests were added.>
This has been running in our production environment for a couple weeks now.

**Documentation:** <Describe the documentation added.>
None.

### CODE REVIEW

File: exporter/coralogixexporter/go.mod [hunk 1]
```diff
@@ -19,7 +19,6 @@ require (
 )
 
 require (
-	cloud.google.com/go/compute/metadata v0.2.4-0.20230617002413-005d2dfb6b68 // indirect
 	github.com/cenkalti/backoff/v4 v4.2.1 // indirect
```
Comment (codeboten) [inline exporter/coralogixexporter/go.mod:22]: Some of these go.mod changes appear unrelated

File: exporter/jaegerexporter/go.mod [hunk 2]
```diff
@@ -23,7 +23,6 @@ require (
 )
 
 require (
-	cloud.google.com/go/compute/metadata v0.2.4-0.20230617002413-005d2dfb6b68 // indirect
```
Comment (codeboten) [inline exporter/jaegerexporter/go.mod:26]: Same here

File: exporter/opencensusexporter/go.mod [hunk 3]
```diff
@@ -22,7 +22,6 @@ require (
 )
 
 require (
-	cloud.google.com/go/compute/metadata v0.2.4-0.20230617002413-005d2dfb6b68 // indirect
```
Comment (codeboten) [inline exporter/opencensusexporter/go.mod:25]: And again

File: receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_windows.go [hunk 4]
```diff
@@ -134,29 +134,29 @@ func (s *scraper) scrape(_ context.Context) (pmetric.Metrics, error) {
 
 func (s *scraper) recordDiskIOMetric(now pcommon.Timestamp, logicalDiskCounterValues []*perfcounters.CounterValues) {
 	for _, logicalDiskCounter := range logicalDiskCounterValues {
-		s.mb.RecordSystemDiskIoDataPoint(now, logicalDiskCounter.Values[readBytesPerSec], logicalDiskCounter.InstanceName, metadata.AttributeDirectionRead)
-		s.mb.RecordSystemDiskIoDataPoint(now, logicalDiskCounter.Values[writeBytesPerSec], logicalDiskCounter.InstanceName, metadata.AttributeDirectionWrite)
+		s.mb.RecordSystemDiskIoDataPoint(now, logicalDiskCounter.Values[readBytesPerSec], logicalDiskCounter.InstanceName, metadata.AttributeDirectionRead, "")
```
Comment (codeboten) [inline receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_windows.go:137]: Does windows not have a similar attribute?

**Review by codeboten — COMMENTED** [review]
Thanks for the contribution @gdvalle! Please add a changelog w/ the enhancement

**Comment by dmitryax** [comment]
This is a significant breaking change and should not be merged in the current form. 

It can be added as an optional resource attribute or as an optional metric attribute after https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16374 is done. 

Also, there is a working group started to stabilize the system metrics https://github.com/orgs/open-telemetry/projects/55. So this change should be reflected in semantic conventions as well

File: exporter/coralogixexporter/go.mod (hunk 1 above)
Comment (gdvalle) [reply exporter/coralogixexporter/go.mod:22]: I believe these all came from running `make gotidy`. Can see the changes isolated in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/26297/commits/eb94c8efadc3e3cb9698fd38065da01deb823c61. Is there another way to get the gopsutil dep update synced without these changes?

**Comment by gdvalle** [comment]
> This is considered a breaking change even if it doesn't increase cardinality and should not be merged in the current form.

For my own understanding, could someone explain what the contract is we're breaking here? I'm asking so I know the acceptance criteria for the next go-round.

I can see hostmetrics labeled as "beta", and from https://github.com/open-telemetry/opentelemetry-collector#beta
> Same as Alpha, but the configuration options are deemed stable. While there might be breaking changes between releases, component owners should try to minimize them. 

Is it then related to the spec described in [semantic-conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/system-metrics.md#systemdisk---disk-controller-metrics), and that gaining a required attribute would violate that? And attribute changes to other receivers, that do not have semantic conventions, would be handled differently?

**Comment by dmitryax** [comment]
> For my own understanding, could someone explain what the contract is we're breaking here? I'm asking so I know the acceptance criteria for the next go-round.

@gdvalle please take a look at this doc. It covers the contract as well

> Is it then related to the spec described in [semantic-conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/system-metrics.md#systemdisk---disk-controller-metrics), and that gaining a required attribute would violate that? And attribute changes to other receivers, that do not have semantic conventions, would be handled differently?

Other receivers are more relaxed, especially those that don't have semantic conventions yet. The system metrics are the most mature at this point, and given that the semantic conventions for them will be marked stable soon, this component has to follow all the definitions. All the attributes have to be defined in semantic conventions as well.

How we go about this PR should be discussed in the System Semantic Conventions working group first. The discussion should be started as an issue in https://github.com/open-telemetry/semantic-conventions/issues. 

To me, this sounds like an optional resource attribute. However, the problem is that the `device` is a metric (not resource) attribute. Maybe we can consider changing that. If we keep the `device` as a metric attribute, this should be added as an optional metric attribute after https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16374 is done.
